File: src/lib/network/protocol/messages/client/ClientHelloMessage.dart
import 'package:sc_utility/network/protocol/Message.dart';
import '../../../../resources.dart';

class ClientHelloMessage extends Message{
  ClientHelloMessage(Resources resources) : super.fromClient(resources){
    id = 10100;
  }

  @override
  void encode() {
    writer.writeInt(2); // Protocol
    writer.writeInt(29); // KeyVersion
    writer.writeInt(3); // Major
    writer.writeInt(0); // Minor
    writer.writeInt(2077); // Build
    writer.writeString(resources.fingerprintSha()); // SHA
    writer.writeInt(2); // Device
    writer.writeInt(2); // AppStore
  }
}

File: src/api/Network/Protocol/Messages/Client/ClientHelloMessage.cs
using SupercellUilityApi.Helpers;
using SupercellUilityApi.Models;

namespace SupercellUilityApi.Network.Protocol.Messages.Client
{
    public class ClientHelloMessage : PiranhaMessage
    {
        public ClientHelloMessage(Network.Client client) : base(client)
        {
            Id = 10100;
        }

        public string Sha { get; set; }
        public GameVersion GameVersion { get; set; }

        public override void Encode()
        {
            Writer.WriteInt(GameVersion.Protocol); // Protocol
            Writer.WriteInt(GameVersion.Key); // KeyVersion
            Writer.WriteInt(GameVersion.Major); // Major
            Writer.WriteInt(GameVersion.Minor); // Minor
            Writer.WriteInt(GameVersion.Build); // Build
            Writer.WriteScString(Sha); // SHA
            Writer.WriteInt(2); // Device
            Writer.WriteInt(2); // AppStore
        }
    }
}